pkg/reconciler/autoscaling/kpa/scaler_test.go

```diff
@@ -175,35 +179,48 @@ func TestScaler(t *testing.T) {
 		t.Run(test.label, func(t *testing.T) {
 			// The clients for our testing.
 			servingClient := fakeKna.NewSimpleClientset()
-			scaleClient := &scalefake.FakeScaleClient{}
+			dynamicClient := fakedynamic.NewSimpleDynamicClient(NewScheme())
 
 			opts := reconciler.Options{
-				ScaleClientSet:   scaleClient,
+				DynamicClientSet: dynamicClient,
 				Logger:           logtesting.TestLogger(t),
 				ConfigMapWatcher: newConfigWatcher(),
 			}
 
 			revision := newRevision(t, servingClient, test.minScale, test.maxScale)
-			deployment := newDeployment(t, scaleClient, names.Deployment(revision), test.startReplicas)
+			deployment := newDeployment(t, dynamicClient, names.Deployment(revision), test.startReplicas)
 			revisionScaler := NewScaler(opts)
 
+			// We test like this because the dynamic client's fake doesn't properly handle
+			// patch modes prior to 1.13 (where vaikas added JSON Patch support).
+			gotScaling := false
+			dynamicClient.PrependReactor("patch", "deployments",
+				func(action clientgotesting.Action) (bool, runtime.Object, error) {
+					patch := action.(clientgotesting.PatchAction)
+					if !test.wantScaling {
+						t.Errorf("don't want scaling, but got patch: %s", string(patch.GetPatch()))
+					}
+					gotScaling = true
+					return true, nil, nil
+				})
+
 			pa := newKPA(t, servingClient, revision)
 			if test.kpaMutation != nil {
 				test.kpaMutation(pa)
 			}
 
 			desiredScale, err := revisionScaler.Scale(logtesting.TestContextWithLogger(t), pa, test.scaleTo)
-
 			if err != nil {
 				t.Error("Scale got an unexpected error: ", err)
 			}
 			if err == nil && desiredScale != test.wantReplicas {
 				t.Errorf("desiredScale = %d, wanted %d", desiredScale, test.wantReplicas)
 			}
 			if test.wantScaling {
-				checkReplicas(t, scaleClient, deployment, test.wantReplicas)
-			} else {
-				checkNoScaling(t, scaleClient)
+				if !gotScaling {
```

Comment:
did we lose the case where we ensure we didn't scale when wantScaling is false?

Reply:
just saw! nm 😅

Reply:
I believe I dealt with the merge confict, but LMK if I screwed it up!